include/rapidjson/document.h

```diff
@@ -2278,17 +2278,18 @@ class GenericDocument : public GenericValue<Encoding, Allocator> {
     //!@name Parse from stream
     //!@{
 
-    //! Parse JSON text from an input stream (with Encoding conversion)
+    //! Parse JSON text from an input stream (with codepoint validate)
     /*! \tparam parseFlags Combination of \ref ParseFlag.
-        \tparam SourceEncoding Encoding of input stream
-        \tparam InputStream Type of input stream, implementing Stream concept
+        \tparam SourceEncoding Encoding of input stream.
+        \tparam CodePointValidation Enable validate or not, implemented by ValidatableEncoder.
+        \tparam InputStream Type of input stream, implementing Stream concept.
         \param is Input stream to be parsed.
         \return The document itself for fluent API.
     */
-    template <unsigned parseFlags, typename SourceEncoding, typename InputStream>
+    template <unsigned parseFlags, typename SourceEncoding, bool CodePointValidation, typename InputStream>
```

Comment:
I thought it was done via `parseFlags`...

Reply:
A new flag? Is there a name recommendation?

Reply:
That may introduce the cost of runtime judgment, so I took it as a second choice at first.

Reply:
Since it is a compile-time parameter, all conditions involving that parameter should be evaluated at compile-time. Actually, I have not analyzed what is the proper way to resolve this. Should `kParseValidateEncodingFlag` include this checking?

Reply:
This will have two effects, the first is that the behavior after triggering a validation failure has changed from throwing an exception to returning a parse error, the second is the performance impact of more checks.
From a macro perspective, I think this behavior becomes more friendly, so I prefer to reuse the flag.
In terms of performance, we may need to observe the impact by means of flame charts, etc. If the performance impact is very obvious, we can create another mark for this check separately.
After having enough data we can explicitly declare this change in the update log and then observe the subsequent feedback.

Reply:
Yes you are right about the conditions evaluation, I was thinking wrong.